libs/EXTERNAL/libezgl/include/ezgl/application.hpp

```diff
@@ -134,6 +144,9 @@ class application {
     : main_ui_resource(m_resource), window_identifier(w_identifier), canvas_identifier(c_identifier), application_identifier(a_identifier),
       setup_callbacks(s_callbacks)
     {
+      // If the default application identifier is used, uniquify it by appending a time stamp
```

Comment:
VPR already uniqifies the application_id:
https://github.com/verilog-to-routing/vtr-verilog-to-routing/blob/8861b1155a5efaa5f8c181063ea36173e9427644/vpr/src/draw/draw.cpp#L227

Doing this all the time will prevent users from using the application uniqueness feature of GTK. Basically, it ensures that there is only one application is running at a time per session (e.g. as in Netbeans). Check [this](https://api.gtkd.org/gio.Application.Application.html). One another benefit of this GTK feature is that you can have multiple processes connected to the same application window. Check "Primary and Remote Instances" [here](http://ssalewski.de/gtkprogramming.html)

I'll add comments explaining that.

Reply:
But if vpr is also uniquifying the id (but in a different way, using pid) then we're always running with a unique application id, but just generate it in two different ways, which is more confusing than always using the same way.
Is that right, or am I missing something?

Reply:
I see your point, but I am treating ezgl as a separate library that can be used by applications other than VTR. So, currently, ezgl has two options:
1. Not to specify the application_id. In this case the default application_id "ezgl.app" is used, and it is appended by a time stamp, so that multiple applications can run at a time (no application uniqueness)
2. Provide an application_id, and in this case the application_id is used as it is. So it is up to the user whether they want the application uniqueness feature or not. In VTR, the application_id is passed as "org.verilogtorouting.vpr.PID#", so multiple vpr instances can run at a time. 

In ezgl, I used time rather than pid to uniquify the id, because I thought it is more portable to use (no C++ standard way to get the pid as far as I know).

Reply:
I see.  I like the time approach (I remember that discussion).  We could change the string passed in to be named application_id_root, and always uniqify it with time and then there would only be one code path.  Could add an optional extra argument `uniqify = true` if you wanted to preserve the ability to set non-unique ids for the caller.
It's more explicit than checking if "ezgl.app" is the passed in name, and removes the OS-dependent pid code path.